### PR TITLE
pythonPackages.fastparquet: 0.5.0 -> 0.6.3

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.fastparquet python39Packages.fastparquet

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124862` at de49efc5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.datashader</li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>python38Packages.cramjam</li>
    <li>python38Packages.fastparquet</li>
    <li>python39Packages.cramjam</li>
  </ul>
</details>
<details>
  <summary>5 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `zstd`.
Near pkgs/development/python-modules/fastparquet/default.nix:2:76:
```
  |
2 | , cramjam, fsspec, thrift, pytestCheckHook, python-snappy, lz4, zstandard, zstd }:
  |                                                                            ^
```
</li>
    <li>warning: unused-argument

Unused argument: `python-snappy`.
Near pkgs/development/python-modules/fastparquet/default.nix:2:45:
```
  |
2 | , cramjam, fsspec, thrift, pytestCheckHook, python-snappy, lz4, zstandard, zstd }:
  |                                             ^
```
</li>
    <li>warning: unused-argument

Unused argument: `zstandard`.
Near pkgs/development/python-modules/fastparquet/default.nix:2:65:
```
  |
2 | , cramjam, fsspec, thrift, pytestCheckHook, python-snappy, lz4, zstandard, zstd }:
  |                                                                 ^
```
</li>
    <li>warning: unused-argument

Unused argument: `lz4`.
Near pkgs/development/python-modules/fastparquet/default.nix:2:60:
```
  |
2 | , cramjam, fsspec, thrift, pytestCheckHook, python-snappy, lz4, zstandard, zstd }:
  |                                                            ^
```
</li>
    <li>warning: unused-argument

Unused argument: `pipBuildHook`.
Near pkgs/development/python-modules/cramjam/default.nix:5:3:
```
  |
5 | , pipBuildHook
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/cramjam/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    cp ${./Cargo.lock} ./Cargo.lock
+  '';
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  nativeBuildInputs = with rustPlatform; [ cargoSetupHook maturinBuildHook ];
+  buildInputs = lib.optional stdenv.isDarwin [ libiconv ];
+  dontUseSetuptoolsBuild = true;
```
Comment (FRidh) [inline pkgs/development/python-modules/cramjam/default.nix:36]: `format = "pyproject";` instead

Comment (veprbl) [reply pkgs/development/python-modules/cramjam/default.nix]: Ok, good

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.fastparquet python39Packages.fastparquet

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.fastparquet python39Packages.fastparquet

**Comment by FRidh** [comment]
@GrahamcOfBorg build python3.pkgs.datashader

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.fastparquet python39Packages.fastparquet
@GrahamcOfBorg build python3.pkgs.datashader

**Comment by veprbl** [comment]
datashader is broken on Hydra https://hydra.nixos.org/build/144120514

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 124862` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.cramjam</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.fastparquet</li>
    <li>python39Packages.cramjam</li>
  </ul>
</details>

**Comment by veprbl** [comment]
Strangely, datashader had built on a second attempt for me.

File: pkgs/development/python-modules/fastparquet/default.nix [hunk 2]
```diff
@@ -1,23 +1,43 @@
...
   nativeBuildInputs = [ pytestrunner ];
-  propagatedBuildInputs = [ numba numpy pandas thrift ];
-  checkInputs = [ pytestCheckHook python-snappy lz4 zstandard zstd ];
+  propagatedBuildInputs = [ cramjam fsspec numba numpy pandas thrift ];
+  checkInputs = [ pytestCheckHook ];
+
+  # Workaround https://github.com/NixOS/nixpkgs/issues/123561
+  preCheck = ''
+    mv fastparquet/test .
+    rm -rf fastparquet
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fastparquet/default.nix:33]:
```suggestion
    rm -r fastparquet
```

File: pkgs/development/python-modules/fastparquet/default.nix [hunk 3]
```diff
@@ -1,23 +1,43 @@
...
+    fastparquet_test="$out"/${python.sitePackages}/fastparquet/test
+    ln -s `pwd`/test "$fastparquet_test"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fastparquet/default.nix:35]:
```suggestion
    ln -s $PWD/test "$fastparquet_test"
```

**Pull request merged**